### ISSUE TITLE
[HOLD for payment 2023-06-05] [$1000] Desktop - Sign in - User can not validate an account

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#17724]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Issue found when executing PR** https://github.com/Expensify/App/pull/17697

## Action Performed:
1. Access the Expensify Desktop app
2. Input an email that does not exist
3. Access the email account and tap on the link (Redirect it to the Desktop app)
4. Input a New password

## Expected Result:
User expects the password to be saved and the new account to be open

## Actual Result:
The user keeps receiving an error message saying that the link has expired, making it impossible to validate an account via Desktop app.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.2.3

**Reproducible in staging?:** Yes

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/233431775-1c0bfa1e-61a7-40db-882f-9ae726c309eb.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~016ce9697b87884431</li>
        <li>Upwork Job ID: 1650976104948047872</li>
        <li>Last Price Increase: 2023-05-11</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @arielgreen (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by mountiny** [comment]
@mvtglobally @kbecciv In the video, you are clicking the link which goes to production, is that not part of the problem? I am not sure

**Comment by MelvinBot** [comment]
@arielgreen Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MelvinBot** [comment]
Triggered auto assignment to @tjferriss (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~016ce9697b87884431

**Comment by MelvinBot** [comment]
Current assignee @tjferriss is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @rushatgabhane (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @bondydaa (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by tjferriss** [comment]
>@mvtglobally @kbecciv In the video, you are clicking the link which goes to production, is that not part of the problem? I am not sure

Following up on @mountiny question.

**Comment by mvtglobally** [comment]
@mountiny @tjferriss the account creation is done in STG app. Should the link be sent as STG as well? Maybe that's the root cause. 

I will ask team to Not click the link, but paste and modify to staging to check

**Comment by mvtglobally** [comment]
Looks to be still repro after URLs switch as well 

https://user-images.githubusercontent.com/43995119/234629323-cdceeca5-b177-4bae-8233-e25f8b594f81.MOV

**Comment by bondydaa** [comment]
https://github.com/Expensify/App/pull/17697 should have nothing to do with sign in links.

I think I see the problem - when you load the magic code in the browser it is essentially "using that code", I'm am not sure if this also then expires that code (I think it does).

So then when you click to allow to open the link in the desktop app it's attempting to use a now expired code and fails.

cc @NikkiWines @johnmlee101 can you help us confirm this or not?

**Comment by johnmlee101** [comment]
Hmm, I'm not sure where in the code do we "use the code" in this case.

Also the validateLink is in the wrong format from what I can tell. https://github.com/Expensify/Web-Expensify/blob/5dc779f8f5ed883805ea035af71d96a590cefef9/lib/Notification/Standard.php#L1529-L1535 @kbecciv can you re-test since we added passwordless to gmail accounts since then. This flow will be removed

**Comment by bondydaa** [comment]
Hmm yeah @hoangzinh that is a good call out. 

I'm not actually sure how Slack handles this, are there browser APIs available to can detect if a link can be opened by another app?

Seems like the flow would need:
1. click link that triggers request in browser
2. (if possible) detect deep/universal link
3. if no deep/universal link then proceed with request to validate
4. if deep/universal link then pass off request/validateCode to that app

If there aren't ways to detect deep/universal links then we might need to come up with a way ourselves within the API to validate/pass through a session from web to the app then, nothing immediately jumps out at me.

**Comment by bondydaa** [comment]
oh interesting... I'm going to comment on that thread and link this issue. 

Going to throw a hold on this for the time being then and we can see what the discussion in slack comes to.

**Comment by bondydaa** [comment]
holding until we can resolve this discussion https://expensify.slack.com/archives/C049HHMV9SM/p1683049880878109?thread_ts=1678823496.930569&cid=C049HHMV9SM

**Comment by tjferriss** [comment]
@bondydaa it looks like we came to a conclusion in Slack. What is the next step here?

**Comment by bondydaa** [comment]
ah thanks for the bump, I had been waiting to see if anyone else was going to weigh in on that thread. 

Taking this off hold then and looking for proposals to implement this: https://expensify.slack.com/archives/C049HHMV9SM/p1683051477869179?thread_ts=1678823496.930569&cid=C049HHMV9SM

> create a navigation route that checks for a `validateCode` code param and then loads a "loading page" like we had before that will allow the user to then open in app or in browser?

**Comment by tjferriss** [comment]
@rushatgabhane @bondydaa what do we think of @hoangzinh proposal?

**Comment by bondydaa** [comment]
I don't think we want to revert the everything from https://github.com/Expensify/App/pull/16383 so I'd say it's best to "re-implement the redirect page only when the route match `ROUTES.VALIDATE_LOGIN`, otherwise we let it render current page behind the Alert."

but otherwise 👍 going to assign you @hoangzinh

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by bondydaa** [comment]
oh wat... did this get fixed elsewhere already?

**Comment by bondydaa** [comment]
okay great thanks for confirming 🙇

**Comment by tjferriss** [comment]
@rushatgabhane looks great.

**Comment by tjferriss** [comment]
Looks like we're still waiting on this PR to see if it fixes the issue https://github.com/Expensify/App/pull/19819

**Comment by bondydaa** [comment]
sorry i haven't really been following 100% here but didn't we merge the deploy the code here? did it get reverted or is it breaking something?

If not let's get payment done then and we can spin up new issues if needed to address other things.

**Comment by tjferriss** [comment]
Great, I think we're all in agreement. We're going to process the payment and close this issue. @rushatgabhane @hoangzinh I've sent both of you an offer in Upworks.

**Comment by tjferriss** [comment]
@rushatgabhane @hoangzinh I've sent you both offers in Upworks.

**Comment by tjferriss** [comment]
The payments have been made in Upworks.

**Comment by tjferriss** [comment]
@hoangzinh it now looks like the contract has ended.

**Comment by tjferriss** [comment]
I think I found it now and ended the contract.
